st3/sublime_lib/output_panel.py

```diff
@@ -2,12 +2,23 @@
 
 
 class OutputPanel(ViewStream):
-    def __init__(self, window, name):
+    def __init__(self, window, name, *,
+        settings=None,
+        read_only=None
+    ):
         super().__init__(window.get_output_panel(name))
 
         self.window = window
         self.name = name
 
+        if settings is not None:
+            view_settings = self.view.settings()
+            for key, value in settings.items():
+                view_settings.set(key, value)
+
+        if read_only is not None:
+            view.set_read_only(read_only)
```

Comment:
In my implementation, I used to unset read-only for the methods that operate on the contents. I believe this is in order here as well, or do you expect the user to handle that manually? In that case I'd rather have them manage read-only entirely by themselves from outside of the class's implementation.